pages/api/webhooks/order-calculate-taxes.ts

```diff
@@ -21,15 +22,24 @@ const handler: NextApiHandler = async (request, response) => {
       .json({ success: false, message: error.message });
     return;
   }
+  const saleorDomain = request.headers[SALEOR_DOMAIN_HEADER];
 
   const body: OrderPayload[] =
     typeof request.body === "string" ? JSON.parse(request.body) : request.body;
 
   const orderPayload: OrderPayload = body[0];
 
-  // FIXME: this part of settings will be fetched from App.metadata and defined based
-  // on channnel used in order.
-  const taxJarConfig = getTaxJarConfig();
+  const taxJarConfig = await getTaxJarConfig(
+    saleorDomain as string,
+    orderPayload.channel.slug
+  );
+  if (!taxJarConfig) {
+    response
+      .status(404)
+      .json({ success: false, message: "TaxJar is not configured." });
+    console.log("TaxJar is not configured.");
```

Comment:
```suggestion
    console.error("TaxJar is not configured.");
```

Reply:
@witoszekdev Hhm. Are you sure that we should always raise an error when someone didn't configure a taxjar?
I am not sure if calling  response with 404 here is the correct way. Like you could get a payload for channel that is not active for given channel. Example: You're using taxjar for US channel, the rest of channels are not configured and should be skipped. So I don't think that calling error here is the correct way.
Also, I will also create a task for better handling channels that are not active.

Reply:
Oh, ok - I thought it's a global configuration that without it the app won't work at all (like API keys).

If you expect TaxJar to not be configured for certain scenarios, then I think you can return `200` so that Saleor won't retry sending the webhook and leave the console.log as it is 👍🏻

Reply:
I would still return a message that TaxJar was not configured so that observability app will catch this in logs.

Reply:
Yeap, exacly. I already created task for this. Thanks for raising the subject <3